gherkin/connection.feature

```diff
@@ -0,0 +1,57 @@
+@rpc @in-process
+Feature: flagd provider disconnect and reconnect functionality
+
+  Scenario Outline: Connection
+    Given a <name> flagd provider
+    And a Boolean-flag with key "boolean-flag" and a default value "true"
+    When the flag was evaluated with details
+    Then the resolved details value should be "true"
+
+    Examples: Stable
+      | name   |
+      | stable |
+    @targetURI
+    Examples: Target URI
+      | name   |
+      | target |
+    @customCert
+    Examples: Certificates
+      | name |
+      | ssl  |
+    @unixsocket @os.linux
+    Examples: Unixsocket
+      | name   |
+      | socket |
+
+  @reconnect
+  # This test suite tests the reconnection functionality of flagd providers
+  Scenario Outline: Provider reconnection
+    Given a <name> flagd provider
+    And a ready event handler
+    And a error event handler
+    When a ready event was fired
+    When the connection is lost for 4s
+    Then the error event handler should have been executed
+    Then the ready event handler should have been executed
+
+    Examples: Stable
```

Comment:
This is a bit of a nit, but if I understand correctly, we can use `Scenarios` here instead of `Examples` which I think is just clearer (using SSL or a target URI seems more like a specific "scenario" to me than an "example". Example just sounds like example data, but this is a different mode of operation, not just different sample data.

If this is a difficulty, I'm fine with `Example`, but my understanding is they are synonymous in cucumber.